Source/WebCore/platform/network/curl/OpenSSLHelper.cpp

```diff
@@ -301,19 +301,74 @@ static void getSubjectAltName(const X509* x509, Vector<String>& dnsNames, Vector
             if (value->d.iPAddress->length == 4)
                 ipAddresses.append(makeString(data[0], ".", data[1], ".", data[2], ".", data[3]));
             else if (value->d.iPAddress->length == 16) {
-                StringBuilder ipAddress;
-                for (int i = 0; i < 8; i++) {
-                    ipAddress.append(makeString(hex(data[0] << 8 | data[1], 4)));
-                    if (i != 7)
-                        ipAddress.append(":");
-                    data += 2;
-                }
-                ipAddresses.append(ipAddress.toString());
+                ipAddresses.append(canonicalizeIPv6Address(data));
             }
         }
     }
 }
 
+String canonicalizeIPv6Address(unsigned char *data)
```

Comment:
This function signature still does not indicate the length.  I think it would be far superior to use a function signature like this:
String canonicalizeIPV6Address(Span<uint8_t, 16>)
Otherwise, this function should be robust against any length of data being its input.

Reply:
Addressed in https://github.com/WebKit/WebKit/pull/3791/commits/d4f739bfcbb4681199f23013d7ac903f78aff943